backend/coins/coin/amount.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Shift Devices AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coin
+
+import (
+	"math/big"
+	"strings"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
+)
+
+// Amount represents an amount in the smallest coin unit (e.g. satoshi).
+type Amount struct {
+	*big.Int
```

Comment:
@x1ddos 

the reason i put it into the struct was to avoid awkward conversions. in the case of `type Amount big.Int`, converting `*big.Int` to `*Amount` seems impossible without a copy (or am i missing sth?):

```
func NewAmount(amount *big.Int) *Amount {
	tmp := Amount(*amount)
	return &tmp
}
```

In the latter case, I would need to clutter my code with `&amount.Int` to extract `*big.Int`.

>  For instance, when passing a Amount to a func, I would expect a value to be copied. So, whatever happens to a in a function would leave the caller's value in the original state.

Could be mitigated with passing `*Amount` instead?